common/src/main/java/org/apache/uniffle/common/compression/Compressor.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.common.compression;
+
+public interface Compressor {
```

Comment:
Do we need to merge Compressor and Decompressor into one interface "Codec" like hadoop ?
It is more concise and avoid to mix different pair of Compressor and Decompressor.

@jerqi @zuston

Reply:
Let me do a simple review about hadoop codec.

Reply:
I dont find co/decompressor mixed in Hadoop one interface. 

https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/Compressor.java

Reply:
I mean the compress/decompress could share the same interface for the user.
For example, `CompressionCodec` has `createOutputStream` (for compress) and `createInputStream` (for decompress).

Reply:
So you mean that I need to create similar Zstd/LZ4CompressionCodec to implement the Compressor and Decompressor interface?

Reply:
You can only provide a Codec instead of CompressionFactory, which hides the inner compressor and decompressor.
The user could directly use Codec to compress or decompress data, so that the user does not need to use compressor and decompressor directly.

Reply:
I propose the new commit according to your idea, https://github.com/apache/incubator-uniffle/pull/254/commits/866b642342b5854f266e6a03d07c673dc61a2f90

Do I get your point? @frankliee

Reply:
I prefer this style 
```
abstract class Codec {
   private static class Compressor {}
   private static class deCompressor{}
   private getCompressor()  // for init lazily
   private getDeCompressor()
 
   public compress() 
   public deCompress()
}

ZSTDCodec extends Codec {}
LZ4Codec extends Codec {}
```

Reply:
Emm.... OK. I will obey this project style.